packages/labs/src/components/select/select.tsx

```diff
@@ -269,6 +279,12 @@ export class Select<T> extends React.Component<ISelectProps<T>, ISelectState<T>>
             this.list.scrollActiveItemIntoView();
         }
 
+        requestAnimationFrame(() => {
```

Comment:
_Snazzy_. 👏 Though this seems brittle. Can you write a unit test for this to guarantee the behavior? Probably something like:

```tsx
it("should focus input when popover opened if autofocus=true", (done) => {
    // setup, open popover
    // use setTimeout to wait for the focus to transfer within requestAnimationFrame
    setTimeout(() => {
        // check: is popover input focused?
        done();
    });
}); 
```

Reply:
this should happen in the ref handler itself, not on every render.

`autoFocus` only applies when the component appears, not each time it updates.

> This Boolean attribute lets you specify that a form control should have input focus **when the page loads** ([MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input))

Reply:
might be worth respecting `inputProps.autoFocus === false`? I could go either way.

Reply:
Is there a good way to test for this when we use requestAnimationFrame? Was able to write a test that passes in the browser, but not in the cli. Found this comment, which seems related: https://github.com/palantir/blueprint/blob/874b757e729576a330ee3a8b67a382c7ff4a312e/packages/core/test/common/utilsTests.ts#L106-L107

Reply:
just use `setTimeout` to test it on the next frame. 

that `utilsTest` is for a util that must use `rAF`, but for other focus stuff a timeout is enough.